src/outputs/ascent.cpp

```diff
@@ -0,0 +1,203 @@
+    int nz = kb.e - kb.s + 1;
+    uint64_t ncells = nx * ny * nz;
+
+    auto ib_int = bounds.GetBoundsI(IndexDomain::interior);
+    auto jb_int = bounds.GetBoundsJ(IndexDomain::interior);
+    auto kb_int = bounds.GetBoundsK(IndexDomain::interior);
+
+    auto &coords = pmb->coords;
+    Real dx1 = coords.CellWidth<X1DIR>(ib.s, jb.s, kb.s);
+    Real dx2 = coords.CellWidth<X2DIR>(ib.s, jb.s, kb.s);
+    Real dx3 = coords.CellWidth<X3DIR>(ib.s, jb.s, kb.s);
+    std::array<Real, 3> corner = coords.GetXmin();
+
+    // create the coordinate set
+    mesh["coordsets/coords/type"] = "uniform";
+
+    mesh["coordsets/coords/dims/i"] = nx + 1;
+    mesh["coordsets/coords/dims/j"] = ny + 1;
+    if (nz > 1) {
+      mesh["coordsets/coords/dims/k"] = nz + 1;
+    }
+
+    // add origin and spacing to the coordset (optional)
+    mesh["coordsets/coords/origin/x"] = corner[0];
+    mesh["coordsets/coords/origin/y"] = corner[1];
+    if (nz > 1) {
+      mesh["coordsets/coords/origin/z"] = corner[2];
+    }
+
+    mesh["coordsets/coords/spacing/dx"] = dx1;
+    mesh["coordsets/coords/spacing/dy"] = dx2;
+    if (nz > 1) {
+      mesh["coordsets/coords/spacing/dz"] = dx3;
+    }
+
+    // add the topology
+    mesh["topologies/topo/type"] = "uniform";
+    mesh["topologies/topo/coordset"] = "coords";
+
+    // indicate ghost zones with ascent_ghosts set to 1
+    Node &n_field = mesh["fields/ascent_ghosts"];
+    n_field["association"] = "element";
+    n_field["topology"] = "topo";
+    n_field["values"].set(conduit::DataType::int32(ncells)); // NOLINT
+    conduit::int32_array vals_array = n_field["values"].value();
+
+    int idx = 0;
+    for (int k = kb.s; k <= kb.e; k++) {
+      for (int j = jb.s; j <= jb.e; j++) {
+        for (int i = ib.s; i <= ib.e; i++) {
+          if ((i < ib_int.s) || (ib_int.e < i) || (j < jb_int.s) || (jb_int.e < j) ||
+              ((nz > 1) && ((k < kb_int.s) || (kb_int.e < k)))) {
+            vals_array[idx] = 1;
+          } else {
+            vals_array[idx] = 0;
+          }
+          idx++;
+        }
+      }
+    }
+
+    // create a field for each component of each variable pack
+    auto &mbd = pmb->meshblock_data.Get();
+
+    for (const auto &var : mbd->GetCellVariableVector()) {
+      PARTHENON_REQUIRE(var->GetDim(6) * var->GetDim(5) == 1,
+                        "Only scalar and vector fields are currently exported in Ascent. "
+                        "Please open an issue on GitHub if you have a use case.");
+
+      const std::string packname = var->label();
+      auto const &labels = var->metadata().getComponentLabels();
+
+      for (int icomp = 0; icomp < labels.size(); ++icomp) {
+        auto const data = Kokkos::subview(var->data, 0, 0, icomp, Kokkos::ALL(),
+                                          Kokkos::ALL(), Kokkos::ALL());
+        const std::string varname = packname + ":" + labels.at(icomp);
+        mesh["fields/" + varname + "/association"] = "element";
```

Comment:
Assume this will have to change for face-centered fields?

Reply:
Probably more would have to change for face fields.
We could like still use element and Cartesian geometry,but would need to adjust  the locations and element indices, i.e., create a new `topology` and then assign the face fields with that updated topology.

Nevertheless, if you agree with the "loud fail" option I suggested above for tensor variables, I'd add a loud fail here so that we remember to fix this once we have face fields.

Reply:
See above comment: Rather than loud failing we should warn and skip. Otherwise the code will be inoperable with ascent and some variables.

You should also probably check that the variable shape is the same as the meshblock, as it might not be and that would also cause problems. (For example axisymmetric variables.)

Reply:
Oh interesting. Well for now the solution is the same. Skip them. :)

Reply:
I'll leave this as is for now (no warning etc) assuming that the output with Ascent will be obviously wrong if someone is going to use it for non-cell-centered fields.

Reply:
Just so long as ascent doesn't crash or something. We should maybe check that that's the case?

Reply:
I'm at the moment not too concerned about this, given that I'm fairly sure that we'll be the first users of those features once we've implemented them.
And so far if Ascent crashed the backtrace always pointed out to an Ascent function, i.e., straightforward to track down.

Reply:
Ascent natively supports vertex-centered fields: https://ascent.readthedocs.io/en/latest/Tutorial_Intro_Conduit_Blueprint.html?highlight=vertex#creating-a-uniform-grid-with-a-single-field. Just not face-centered ones.